opm/material/common/Tabulated1DFunction.hpp

```diff
@@ -258,7 +258,12 @@ class Tabulated1DFunction
     Evaluation eval(const Evaluation& x, bool extrapolate = false) const
     {
         size_t segIdx = findSegmentIndex_(x, extrapolate);
+        return eval(x,segIdx);
+    }
 
+    template <class Evaluation>
+    Evaluation eval(const Evaluation& x, size_t segIdx) const
```

Comment:
Having two overloads that only differ in integer types may be a little dangerous. If you pass an `int` I think that gets promoted to a `size_t` which is fine. But sometimes an `int` will be used in a `bool` context, and if that is then passed here, then behaviour would change. A hypothetical example:
```
int extrapolationLevel(); // Returns 0 for no extrapolation, 1-3 indicating some kind of extrapolation
table.eval(x, extrapolationLevel()); // Now, the size_t overload will be chosen instead of the bool, giving wrong behaviour.
```
This may be very theoretical, but if @bska or @akva2 could evaluate this risk I would appreciate it.

Reply:
yeah, i agree in principle, this is potentially a nightmare that requires stepping through the debugger to realize. of course your example is rather shitty code, implicit int to bool conversions suck. i don't think this is the only such example though if you look through the codebase.

furthermore isn't the code here an infinite loop for the bool overload?

Reply:
I do not see the infinite loop, would not the call `eval(x, segIdx)` call the size_t overload?

We could modify this to make use a `struct SegmentIndex { size_t value; };` instead of a raw `size_t`, since they are returned by methods here that would be auto-able and not very inconvenient. What do you think?

Reply:
i swear it showed as  return eval(x,extrapolate); last time! it's fine now.

if i understand correctly the goal of this it is being to call the size_t overload directly avoiding the search, so that sounds a bit inconvenient?

Reply:
indeed, the infloop was fixed in [74285d7](https://github.com/OPM/opm-common/pull/3421/commits/74285d79647c235ceaf8f6587f26e5bf706cc6e6)

Reply:
You always need to do a search, but you can use the same segIdx multiple times, saving time used to repeat the search. So there would never be a segment index coming out of nowhere here, it would always come from a findSegmentIndex_ call.

Reply:
done. Should one substitute all the size_t use?

Reply:
I leave this problem. The problem if is someone put unsigned in stead of size_t.